watchmal/engine/engine_regression.py

```diff
@@ -0,0 +1,555 @@
+
+    def configure_data_loaders(self, data_config, loaders_config, is_distributed, seed):
+        """
+        Set up data loaders from loaders config
+
+        Args:
+            data_config     ... hydra config specifying dataset
+            loaders_config  ... hydra config specifying dataloaders
+            is_distributed  ... boolean indicating if running in multiprocessing mode
+            seed            ... seed to use to initialize dataloaders
+
+        Parameters:
+            self should have dict attribute data_loaders
+        """
+        for name, loader_config in loaders_config.items():
+            self.data_loaders[name] = get_data_loader(**data_config, **loader_config,
+                is_distributed=is_distributed, seed=seed)
+
+    def get_synchronized_metrics(self, metric_dict):
+        """
+        Gathers metrics from multiple processes using pytorch distributed operations
+
+        Args:
+            metric_dict         ... dict containing values that are tensor outputs of a single process
+
+        Returns:
+            global_metric_dict  ... dict containing concatenated list of tensor values gathered from all processes
+        """
+        global_metric_dict = {}
+        for name, array in zip(metric_dict.keys(), metric_dict.values()):
+            tensor = torch.as_tensor(array).to(self.device)
+            global_tensor = [torch.zeros_like(tensor).to(self.device) for i in range(self.ngpus)]
+            torch.distributed.all_gather(global_tensor, tensor)
+            global_metric_dict[name] = torch.cat(global_tensor)
+
+        return global_metric_dict
+
+    def forward(self, train=True):
+        """
+        Compute predictions and metrics for a batch of data
+
+        Args:
+            train   ... whether to compute gradients for backpropagation
+
+        Parameters:
+            self should have attributes data, labels, model, criterion, softmax
+
+        Returns:
+            dict containing loss, and model outputs
+        """
+
+        with torch.set_grad_enabled(train):
+            # Move the data and the labels to the GPU (if using CPU this has no effect)
+            self.data = self.data.to(self.device)
+            self.energies = self.energies.to(self.device)
+            self.positions = torch.squeeze(self.positions).to(self.device)
+            loss = None
+            if self.output_type == 'position':
+                loss = self.scale_positions(self.positions).to(self.device)
+            elif self.output_type == 'energies':
+                loss = self.fit_transform(self.energies).to(self.device)
+            model_out = self.model(self.data)
+
+            self.loss = self.criterion(model_out, loss)
+
+        return {'loss': self.loss.detach().cpu().item(),
+                'output': model_out.detach().cpu().numpy(),
+                'raw_output': model_out}
+
+    def scale_positions(self, data):

```

Comment:
- While re-running the file (as is) we faced an error of "invalid index" exactly after 1 whole epoch iterated.
-  Are these the same files used to report the findings in the final report? 
- We also got an error when we tried to train only with energies. 
- The scaling is used in the code as well (below). The scaling was not mentioned in any report. Is this something newly added or previously mentioned elsewhere?
@youngk313

Reply:
<img width="816" alt="Screenshot 2021-05-26 at 12 15 48 AM" src="https://user-images.githubusercontent.com/54958228/119552254-1e01d200-bdb8-11eb-9b45-08d2f455d7b1.png">

Reply:
Hmm that is strange. I just ran it using positions today and I had no issues. As for the scaling, it was mentioned in the final report that is mentioned here: https://www.watchmal.org/projects/regression/bcit-meetings-2021. I will try running with energy and let you know if I encounter the same errors.

Reply:
Thanks for checking this. Can you confirm you're using exactly the version of the code in this pull request, and using the config that is the same as in this pull request?